scripts/west_commands/run_common.py

```diff
@@ -161,6 +161,17 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
     # flashing specific boards prior to others.
     board_priority = []
 
+    # Holds a list of board image flash counts , the first element is number
+    # of images flashed so far and second element is total number of images
+    # for a given board.
+    board_image_count = {}
+
+    # Holds a dictionary of boards that should be prevented from being reset
+    # if there are multiple images to flash. This allows for flashing of
+    # images that e.g. setup security but allow for the remaining images to
+    # be flashed before that protection is enabled.
+    board_reset = []
```

Comment:
Wonder if this should be turned around, that is that when flashing multiple images then system will not be reset until final image has been flashed.

If a reset is required after other images, then the generated `runners.yaml` can contain this information in the runner specific arguments section.

I believe that would require fewer changes.

Reply:
> it would actually just be better if all flash runners could run without doing any resetting, then once done, all cores are reset with a final invocation of each flash runner

that was exactly what I was saying, remove the auto reset we have today: `that is that when flashing multiple images then system will not be reset until final image has been flashed.`

if, for some unforeseen reason, a given image requires a reset, then it would be possible to use the function `board_runner_args(<runner> "--erase")` to forcefully request a reset for that image, even if reset would not be invoked.

Pros:
- No reset after flashing of images
- No need to keep a list of boards that should not be reset (cause there is only a single reset after all images are flashed)
- Possibility for an image to forcefully request a reset (already supported today) to accommodate special needs we don't know about atm.

If images must be flashed in a specific order, then that is a dependency tree the build system must know about, not the flash runner.